### PR TITLE
Fixed Fray, Added Statue for Fray, Added Armor Duraility  and Random Unravled Fabric, Fixed Limbo Teleportation, Other Stuff.

### PR DESCRIPTION
**Pull request by MalekiRe**
Inverted fray so that way incrementing fray actually increments it.
Added a debug command "fray" which allows you to simulate adding any amount of fray to yourself.
Added the Fray enchantment that applies to tools and armor if your in limbo for long enough, randomly.
Added the String Theory enchantment that applies to armor and reduces fray gain by 15% per armor piece, multiplicative, not additive.
A statue spawns and the player dies and fray is reset when u hit over 200 fray.
Over 100 fray has different effects that occur at different levels.
Armor randomly takes durability damage.
Your inventory randomly gets filled with unravled fabric.
at over 175 fray a static vingette comes into focus based on the fray level until its at 200.
When you use the dimteleport command or go through a rift, your uuid is stored in the DimensionalRegistry.
When you exit limbo through a reality liquid, or rift now you don't get teleported to the bottom of limbo and die, instead you get teleported to around the area that you entered the last entrance rift, or if you used dim teleport last, the last dimteleport location you teleported from.
Unravled fabric blocks take time to break instead of being insta-mine.
Advancments are fixed so 3 of them don't trigger when u first get any item, now only triggers when u actually are supposed to trigger it.

### CODE REVIEW

**Review by Waterpicker — APPROVED** [review]

File: src/main/java/org/dimdev/dimdoors/entity/ai/MonolithAggroGoal.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public void tick() {
 				this.target.teleport(this.target.getX(), this.target.getY() + 256, this.target.getZ());
                 this.target.world.playSound(null, new BlockPos(this.target.getPos()), ModSoundEvents.CRACK, SoundCategory.HOSTILE, 13, 1);
                 this.target.incrementStat(ModStats.TIMES_TELEPORTED_BY_MONOLITH);
-				PlayerModifiersComponent.incrementFray(this.target, DimensionalDoorsInitializer.getConfig().getPlayerConfig().fray.monolithTeleportationIncrement);
+				PlayerModifiersComponent.incrementFray(this.target, -DimensionalDoorsInitializer.getConfig().getPlayerConfig().fray.monolithTeleportationIncrement);
```
Comment (CreepyCre) [inline src/main/java/org/dimdev/dimdoors/entity/ai/MonolithAggroGoal.java:111]: Why subtract the increment? Afaik every config value for fray change should be understood as an increment, but most would be negative. This is to ensure consistency of config options.

File: src/main/java/org/dimdev/dimdoors/mixin/PlayerEntityMixin.java [hunk 2]
```diff
@@ -18,18 +21,73 @@
 import net.minecraft.util.Identifier;
 import net.minecraft.world.World;
 
+import java.util.Random;
+
 @Mixin(value = PlayerEntity.class, priority = 900)
 public abstract class PlayerEntityMixin extends LivingEntity {
+	private static final int RANDOM_ACTION_BOUND = 400;
+	private static final int CHANCE_TO_DECREASE_ARMOR_DURABILITY = 20;
```
Comment (CreepyCre) [inline src/main/java/org/dimdev/dimdoors/mixin/PlayerEntityMixin.java]: Probabilities should probably be between 0-1, with your current method you may only achieve 1/n probabilities.

File: src/main/java/org/dimdev/dimdoors/mixin/PlayerEntityMixin.java [hunk 3]
```diff
@@ -18,18 +21,73 @@
...
+	Random random = new Random();
+
 	@Shadow
 	public abstract void incrementStat(Identifier stat);
 
 	public PlayerEntityMixin(EntityType<? extends LivingEntity> entityType, World world) {
 		super(entityType, world);
 	}
 
+	@Inject(method = "tick", at = @At("HEAD"), cancellable = true)
```
Comment (CreepyCre) [inline src/main/java/org/dimdev/dimdoors/mixin/PlayerEntityMixin.java]: Move all of your new code into ServerPlayerEntityMixin instead and sync item/ slot changes via Packets. Anything that is random should happen exclusively server-side and be synced to the client.

File: src/main/java/org/dimdev/dimdoors/mixin/PlayerEntityMixin.java [hunk 4]
```diff
@@ -18,18 +21,73 @@
...
+				break;
+			case 1:
+				addRandomUnravledFabric((PlayerEntity) player);
+				break;
+			default:
+		}
+
+	}
+
+	private void addRandomUnravledFabric(PlayerEntity player) {
```
Comment (CreepyCre) [inline src/main/java/org/dimdev/dimdoors/mixin/PlayerEntityMixin.java]: Typo.
Unrav**e**led
Fix in other places as well.

File: src/main/java/org/dimdev/dimdoors/entity/ai/MonolithAggroGoal.java (hunk 1 above)
Comment (MalekiRe) [reply src/main/java/org/dimdev/dimdoors/entity/ai/MonolithAggroGoal.java:111]: ah yes, i didn't catch this, idk why it was negative in the first place

File: src/main/java/org/dimdev/dimdoors/mixin/PlayerEntityMixin.java (hunk 3 above)
Comment (MalekiRe) [reply src/main/java/org/dimdev/dimdoors/mixin/PlayerEntityMixin.java]: i uh, i changed over the stuff, but I didn't have to use packets?? idunno if I did it correctly or not, I don't know how packets or server side stuff works, but I just called the functions in the serverplayerentity mixin and everything still works?

**Pull request merged**